pkg/rancher-desktop/backend/containerClient/mobyClient.ts

```diff
@@ -278,25 +286,55 @@ export class MobyClient implements ContainerEngineClient {
         }));
         break;
       case 'file': case 'contiguous-file': {
-        await Promise.all(linkNames.map(async(linkName) => {
+        const fileNames = linkNames.map((linkName) => {
           const fileName = absPath(linkName);
 
           if (!fileName) {
             console.warn(`Skipping unexpected file ${ entry.header.name } -> ${ linkName }`);
+          }
 
-            return;
+          return fileName;
+        }).filter(defined);
+
+        await Promise.all(fileNames.map((fileName) => {
+          return fs.promises.mkdir(path.dirname(fileName), { recursive: true });
+        }));
+
+        const writers = fileNames.map(f => fs.createWriteStream(f));
+
+        entry.on('data', async(chunk) => {
+          entry.pause();
+          try {
+            await Promise.all(writers.map(async(writer) => {
+              await new Promise<void>((resolve, reject) => {
+                writer.write(chunk, 'utf-8', (error) => {
```

Comment:
`createReadStream` uses `null` by default (so we should always get `Buffer`s, and not strings); this just says `utf-8` do have some default.  But yes, `binary` would make more sense (but either way it's unused, since we get `Buffer`s, but TypeScript doesn't know that).